components/dashboard/src/components/IdeOptions.tsx

```diff
@@ -0,0 +1,283 @@
+                    <Button variant="secondary" onClick={onClose}>
+                        Cancel
+                    </Button>
+                    <LoadingButton
+                        disabled={!!computedError || props.isLoading}
+                        type="submit"
+                        loading={updateMutation.isLoading}
+                    >
+                        Save
+                    </LoadingButton>
+                </div>
+            </ModalBaseFooter>
+        </Modal>
+    );
+};
+
+interface IdeOptionSwitchProps {
+    ideOption: IDEOption & { id: string };
+    ideVersions: string[] | undefined;
+    pinnedIdeVersion: string | undefined;
+    checked: boolean;
+    onPinnedIdeVersionChange: (version: string | undefined) => void;
+    onCheckedChange: (checked: boolean) => void;
+}
+const IdeOptionSwitch = ({
+    ideOption,
+    ideVersions,
+    pinnedIdeVersion,
+    checked,
+    onCheckedChange,
+    onPinnedIdeVersionChange,
+    ...props
+}: IdeOptionSwitchProps) => {
+    const label = (
+        <>
+            <img className="w-8 h-8 self-center mr-2" src={ideOption.logo} alt="" />
+            <span className="font-medium text-pk-content-primary">{ideOption.title}</span>
+        </>
+    );
+
+    const versionSelector = !!pinnedIdeVersion && !!ideVersions && (
+        <select
+            className="py-0 pl-2 pr-7 w-auto"
+            name="Editor version"
+            value={pinnedIdeVersion}
+            onChange={(e) => onPinnedIdeVersionChange(e.target.value)}
+        >
+            {ideVersions.map((v) => (
+                <option key={v} value={v}>
+                    {v}
+                </option>
+            ))}
+        </select>
+    );
+    const description = (
+        <div className="inline-flex text-pk-content-primary">
+            {versionSelector ? (
+                <>
+                    <MiddleDot />
+                    {versionSelector}
+                </>
+            ) : (
+                ideOption.imageVersion && (
+                    <>
+                        <MiddleDot />
+                        <span className="text-pk-content-primary">{ideOption.imageVersion}</span>
+                    </>
+                )
+            )}
+            <MiddleDot />
+            <span className="text-pk-content-primary capitalize">{ideOption.type}</span>
+        </div>
+    );
+    return (
+        <div className="flex w-full justify-between items-center mt-2">
+            <SwitchInputField
+                key={ideOption.id}
+                id={ideOption.id}
+                label={label}
+                description={description}
+                labelLayout="row"
+                checked={checked}
+                onCheckedChange={onCheckedChange}
+                title={ideOption.title}
+            />
+            {ideOption.pinnable && ideVersions ? (
+                <Button
+                    type="button"
+                    onClick={(e) => {
```

Comment:
playing the linter here a bit :P
```suggestion
                    onClick={() => {
```